pkg/frontend/openshiftcluster_preflightvalidation_test.go

```diff
@@ -0,0 +1,140 @@
+		wantError        string
+		wantResponse     *api.ValidationResult
+	}
+	for _, tt := range []*test{
+		{
+			name: "Successful Preflight",
+			fixture: func(f *testdatabase.Fixture) {
+				f.AddSubscriptionDocuments(&api.SubscriptionDocument{
+					ID: mockSubID,
+					Subscription: &api.Subscription{
+						State: api.SubscriptionStateRegistered,
+						Properties: &api.SubscriptionProperties{
+							TenantID: "11111111-1111-1111-1111-111111111111",
+						},
+					},
+				})
+			},
+			preflightRequest: func() *api.PreflightRequest {
+				return &api.PreflightRequest{
+					Resources: []json.RawMessage{
+						[]byte(`
+								{
+									"apiVersion": "2022-04-01",
+									"name": "resourceName",
+									"type": "microsoft.redhatopenshift/openshiftclusters",
+									"location": "eastus",
+									"properties": {
+										"clusterProfile": {
+										  "domain": "example.aroapp.io",
+										  "resourceGroupId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resourcename",
+										  "fipsValidatedModules": "Enabled"
+										},
+										"consoleProfile": {},
+										"servicePrincipalProfile": {
+										  "clientId": "00000000-0000-0000-1111-000000000000",
+										  "clientSecret": "00000000-0000-0000-0000-000000000000"
+										},
+										"networkProfile": {
+										  "podCidr": "10.128.0.0/14",
+										  "serviceCidr": "172.30.0.0/16"
+										},
+										"masterProfile": {
+										  "vmSize": "Standard_D32s_v3",
+										  "subnetId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/ms-eastus/providers/Microsoft.Network/virtualNetworks/dev-vnet/subnets/CARO2-master",
+										  "encryptionAtHost": "Enabled",
+										  "diskEncryptionSetId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/ms-eastus/providers/Microsoft.Compute/diskEncryptionSets/ms-eastus-disk-encryption-set"
+										},
+										"workerProfiles": [
+										  {
+											"name": "worker",
+											"vmSize": "Standard_D32s_v3",
+											"diskSizeGB": 128,
+											"subnetId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/ms-eastus/providers/Microsoft.Network/virtualNetworks/dev-vnet/subnets/CARO2-worker",
+											"count": 3,
+											"encryptionAtHost": "Enabled",
+											"diskEncryptionSetId": "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/ms-eastus/providers/Microsoft.Compute/diskEncryptionSets/ms-eastus-disk-encryption-set"
+										  }
+										],
+										"apiserverProfile": {
+										  "visibility": "Public"
+										},
+										"ingressProfiles": [
+										  {
+											"name": "default",
+											"visibility": "Public"
+										  }
+										]
+									  }
+								}
+						`),
+					},
+				}
+			},
+			wantStatusCode: http.StatusOK,
+			wantResponse: &api.ValidationResult{
+				Status: api.ValidationStatusSucceeded,
+			},
+		},
+	} {
```

Comment:
can you add a test case with template expression?

Reply:
As discussed, will do this after we test on int so that we can see the preflight payload.